src/component/GoodsList.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+type State = {
+  isReversed: boolean;
+  sortBy: string;
+};
+
+type Props = {
+  goods: string[];
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    isReversed: false,
+    sortBy: '',
+  };
+
+  reverse = () => {
+    this.setState(currentState => ({
+      isReversed: !currentState.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  getVisibleGoods = () => {
+    const { goods } = this.props;
+    const { isReversed, sortBy } = this.state;
+
+    let visibleGoods = goods;
```

Comment:
```suggestion
    let visibleGoods = [...goods];
```
then you no need to make a copy in all places